nms/app/packages/magmalte/app/state/TraceState.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import MagmaV1API from '@fbcnms/magma-api/client/WebClient';
+
+import type {
+  call_trace,
+  call_trace_config,
+  mutable_call_trace,
+  network_id,
+} from '@fbcnms/magma-api';
+
+type InitTraceStateProps = {
+  networkId: network_id,
+  setTraceMap: ({[string]: call_trace}) => void,
+  enqueueSnackbar?: (msg: string, cfg: {}) => ?(string | number),
+};
+
+export async function InitTraceState(props: InitTraceStateProps) {
+  const {networkId, setTraceMap, enqueueSnackbar} = props;
+  try {
+    const traces = await MagmaV1API.getNetworksByNetworkIdTracing({
+      networkId,
+    });
+    if (traces) {
+      setTraceMap(traces);
+    }
+  } catch (e) {
+    enqueueSnackbar?.('failed fetching call trace information', {
+      variant: 'error',
+    });
+    return;
+  }
+}
+
+type CallTraceProps = {
+  networkId: network_id,
+  callTraces: {[string]: call_trace},
+  setCallTraces: ({[string]: call_trace}) => void,
+  key: string,
+  value?: call_trace_config | mutable_call_trace,
```

Comment:
$Shape< call_trace_config & mutable_call_trace > 
https://flow.org/en/docs/types/utilities/#toc-shape and avoid the flow suppress comment

Reply:
Oh that's very cool